users/serializers.py

```diff
@@ -43,6 +45,13 @@ def validate_email(self, email: str) -> str:
                 raise serializers.ValidationError(CustomValidationErrorText.VALIDATION_ERROR_SIGNUP_EMAIL_MESSAGE)
         return email
 
+    def validate_date_of_birth(self, date_of_birth: datetime.date) -> datetime.date:  # pylint: disable=no-self-use
```

Comment:
This should be done directly on `CustomUser.date_of_birth` field as `validators=[MaxValidator(), MinValidator]` (https://docs.djangoproject.com/en/2.2/ref/validators/#maxvaluevalidator). This way we control it on lowest possible level in the app and don't have to worry about it anywhere else. If for some reason it won't work there, then `clean()` method of the model. The case is that we want to remove all remaining of `django-rest-framework` so this serializer will be removed soon, and we shouldn't loose this logic.

Reply:
@rwrzesien  Ok, I will move it, but ther's a problem with `MinValueValidator / MaxValueValidator`. Namely, the min and max value are not set in stone, they depend on the current date (for example, if yesterday user was not 18 years old validation should rasie exception, but if today user turned 18, it should pass).

I will create a custom validator and use it in model.